File: fix_urls.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(dirPath);
  });
}

const targetDirs = ['app', 'lib'];

targetDirs.forEach(dir => {
  walkDir(dir, (filePath) => {
    if (filePath.endsWith('.ts') || filePath.endsWith('.tsx')) {
      let content = fs.readFileSync(filePath, 'utf8');
      if (content.includes('https://fizikhub.com')) {
        let newContent = content.replace(/https:\/\/fizikhub\.com/g, 'https://www.fizikhub.com');
        fs.writeFileSync(filePath, newContent, 'utf8');
        console.log(`Updated: ${filePath}`);
      }
    }
  });
});

File: fix_urls.py
import os

def replace_in_file(filepath):
    with open(filepath, 'r') as f:
        content = f.read()
    if 'https://fizikhub.com' in content:
        content = content.replace('https://fizikhub.com', 'https://www.fizikhub.com')
        # However, if it became https://www.www.fizikhub.com, fix it:
        content = content.replace('www.www.', 'www.')
        with open(filepath, 'w') as f:
            f.write(content)
        print(f"Updated {filepath}")

for root, dirs, files in os.walk('app'):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            replace_in_file(os.path.join(root, file))

for root, dirs, files in os.walk('lib'):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            replace_in_file(os.path.join(root, file))
